controllers/feeds/getFeedsV2.js

```diff
@@ -249,6 +253,13 @@ module.exports = async (req, res) => {
         });
       }
     }
+    // get karma score for each post actor
+    const karmaScores = await UsersFunction.getUsersKarmaScore(User, postActors);
+    for (let i = 0; i < data.length; i++) {
+      const user = karmaScores.find((user) => user.user_id === data[i].actor.id);
+      data[i].karma_score = roundingKarmaScore(user?.karma_score || 0);
+      data[i] = await modifyAnonimityPost(data[i], isBlurredPost);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Fetching karma scores for all actors at once and then mapping them to the posts is a good approach for reducing database queries. However, the current implementation uses a nested loop, resulting in a time complexity of O(n^2). Consider converting `karmaScores` into a Map for faster lookups.

```diff
- const karmaScores = await UsersFunction.getUsersKarmaScore(User, postActors);
+ const karmaScores = new Map((await UsersFunction.getUsersKarmaScore(User, postActors)).map(user => [user.user_id, user]));

for (let i = 0; i < data.length; i++) {
-   const user = karmaScores.find((user) => user.user_id === data[i].actor.id);
+   const user = karmaScores.get(data[i].actor.id);
    data[i].karma_score = roundingKarmaScore(user?.karma_score || 0);
    data[i] = await modifyAnonimityPost(data[i], isBlurredPost);
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->